manifest/build.go

```diff
@@ -21,7 +21,16 @@ func (m *Manifest) Build(dir, appName string, s Stream, cache bool) error {
 		}
 	}
 
+	buildCache := map[string]string{}
+
 	for _, service := range builds {
+		if bc, ok := buildCache[service.Build.Hash()]; ok {
```

Comment:
I'm not convinced Hash() is necessary 

map[manifest.Build]string{} would probably work